sdk/src/main/java/com/mapbox/maps/renderer/MapboxWidgetRenderer.kt

```diff
@@ -117,6 +118,7 @@ internal class MapboxWidgetRenderer(
     }
     this.eglSurface = null
     this.eglCore = null
+    this.eglPrepared = false
```

Comment:
does this actually fix something? wonder how it was working before - if I read the code correctly it meant that we were not calling 
```
if (!eglPrepared) {
      eglPrepared = eglCore.prepareEgl()
      if (!eglPrepared) {
        logE(TAG, "Widget EGL was not configured, please check logs above.")
        return
      }
    }
```

Reply:
I agree that we should set it to false here but I wonder if it's related to current fix. We need to be careful not to break anything 😄

Reply:
Without this, we crash with
```
11-28 11:23:27.037 14947 14994 E AndroidRuntime: FATAL EXCEPTION: MapboxRenderThread
11-28 11:23:27.037 14947 14994 E AndroidRuntime: Process: com.mapbox.maps.testapp, PID: 14947
11-28 11:23:27.037 14947 14994 E AndroidRuntime: kotlin.UninitializedPropertyAccessException: lateinit property egl has not been initialized
11-28 11:23:27.037 14947 14994 E AndroidRuntime: 	at com.mapbox.maps.renderer.egl.EGLCore.createOffscreenSurface(EGLCore.kt:149)
11-28 11:23:27.037 14947 14994 E AndroidRuntime: 	at com.mapbox.maps.renderer.MapboxWidgetRenderer.checkEgl(MapboxWidgetRenderer.kt:187)
11-28 11:23:27.037 14947 14994 E AndroidRuntime: 	at com.mapbox.maps.renderer.MapboxWidgetRenderer.updateTexture(MapboxWidgetRenderer.kt:127)
11-28 11:23:27.037 14947 14994 E AndroidRuntime: 	at com.mapbox.maps.renderer.MapboxRenderThread.draw(MapboxRenderThread.kt:271)
11-28 11:23:27.037 14947 14994 E AndroidRuntime: 	at com.mapbox.maps.renderer.MapboxRenderThread.doFrame(MapboxRenderThread.kt:485)
```

Reply:
@yunikkk as you were writing an the initial logic - could you please take a 👀 .

Reply:
@kiryldz talked offline - this flag should be set here with the current `release` implementation, but it will lead to the GL context being reset on every switch to BG. 
I'll drop a ticket to sync with the main Renderer, where we do not reset the context until renderer is fully destroyed.